lib/commands/source.js

```diff
@@ -15,77 +14,18 @@ commands.getPageSource = async function () {
 };
 
 helpers.getNativePageSource = async function () {
-  let method = 'POST';
-  let endpoint = `/source`;
+  let method = 'GET';
+  let endpoint = `/source/xml`;
   let srcTree = await this.proxyCommand(endpoint, method);
 
-  // translate the returned value into what Appium expects
-  let src = getSourceXml(getTreeForXML(srcTree));
+  // need to add <AppiumAUT /> tags, after the xml header
+  //   <?xml version="1.0" encoding="UTF-8"?>
+  let src = srcTree.tree;
+  let match = XML_HEADER_REGEXP.exec(src);
```

Comment:
A kitten dies somewhere when one parses xml with regexp. Why not to use the standard DOM parser to modify the document?